agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/processors/ExampleExporter.java

```diff
@@ -0,0 +1,342 @@
+                                        return;
+                                    case "hash":
+                                        if(existingSpanAttributes.get(actionObj.key)!=null) {
+                                            AttributeValue existingValue = existingSpanAttributes.get(actionObj.key);
+                                            // Currently we only support String
+                                            if(existingValue.getType() == Type.STRING) {
+                                                builder.setAttribute(actionObj.key, getSHA1(existingValue.getStringValue()));
+                                                updatedFlag=true;
+                                            }
+                                        }
+                                        break;
+                                }
+                            }
+                        }
+                        if(!updatedFlag) {
+                            builder.setAttribute(key, value);
+                        }
+                    }
+                });
+                // loop through insert actions, if key is not in keys set then call builder.setAttribute()
+                return new MySpanData(span, builder.build());
+            }
+        }
+        return span;
+    }
+
+    private String getSHA1(String value) {
+        MessageDigest mDigest = null;
+        try {
+            mDigest = MessageDigest.getInstance("SHA1");
+            byte[] result = mDigest.digest(value.getBytes());
+            StringBuffer sb = new StringBuffer();
+            for (int i = 0; i < result.length; i++) {
+                sb.append(Integer.toString((result[i] & 0xff) + 0x100, 16).substring(1));
+            }
+            return sb.toString();
+        } catch (NoSuchAlgorithmException e) {
+            return value;
+        }
+    }
+
+    private boolean checkExcludes(SpanProcessorIncludeExclude excludes, ReadableAttributes existingSpanAttributes, String spanName) {
+        if(excludes == null || excludes.match_type==null) return false;
+        else if(excludes.match_type.equalsIgnoreCase("strict")) {
+            boolean matchFound = false;
+            if(excludes.span_names!=null && excludes.span_names.size()>0) {
+                if(excludes.span_names.contains(spanName)) {
+                    //match found
+                    matchFound = true;
+                }
+            } else {
+                // if span_names is empty default to true
+                matchFound = true;
+            }
+
+            if(excludes.attributes!=null && excludes.attributes.size()>0) {
+                for(SpanProcessorAttribute attribute:excludes.attributes) {
+                    if(matchFound && attribute.key!=null) {
+                        //found a match
+                        if(existingSpanAttributes.get(attribute.key)!=null) {
+                            if(attribute.value!=null) {
```

Comment:
conditional not needed if we normalize in the "optimized" structure

in optimized structure, remove attribute completely if has no key